synapse/lib/snap.py

```diff
@@ -1380,6 +1380,10 @@ async def addNode(self, name, valu, props=None, norminfo=None):
             mesg = 'The snapshot is in read-only mode.'
             raise s_exc.IsReadOnly(mesg=mesg)
 
+        form = self.core.model.reqForm(name)
```

Comment:
Minor nit but seems like we could move this inside the `isinstance(valu, dict)`  block to avoid grabbing the form twice since `editor.addNode` will grab it again anyways.